packages/flutter_tools/gradle/module_plugin_loader.gradle

```diff
@@ -35,6 +37,17 @@ if (pluginsFile.exists()) {
 
 gradle.getGradle().projectsLoaded { g ->
     g.rootProject.beforeEvaluate { p ->
+        //let the plugin's output generated in flutterModule/build/host/${androidPlugin.name}.
+        p.subprojects { sp ->
```

Comment:
Just to clarify, Gradle subprojects don't equal Flutter plugins always. 
Flutter plugins is a subset of Gradle subprojects.

If you want to process the plugins only, move the logic below to line 34.  See `project(":${androidPlugin.name}")`, that is the subproject you are looking for.

Reply:
@blasten Thanks for your advise. I have tried it just moment. But I cannot modify the build directory use `project(":${androidPlugin.name}")`. `project(":${androidPlugin.name}")` returns the class `org.gradle.initialization.DefaultProjectDescriptor`, It's only allowed me to modify "source dir" ,"(sub)project name" and "(sub)project build script file" . you can see [this doc](https://docs.gradle.org/current/javadoc/org/gradle/api/initialization/ProjectDescriptor.html). 
and `gradle.getGradle().projectsLoaded` returns the class 'org.gradle.api.internal.project.DefaultProject_Decorated' ,It is allowed to modify the build directory.

Reply:
In this case can we add comments?